examples/bpi-monolith-test-suite/bpi/components/zkcircuits/circuitlib/orderCircuit.go

```diff
@@ -0,0 +1,182 @@
+package circuitlib
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"zkcircuit/json"
+
+	"github.com/consensys/gnark-crypto/ecc"
+	"github.com/consensys/gnark/backend"
+	"github.com/consensys/gnark/backend/groth16"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/hash/mimc"
+)
+
+type OrderCircuit struct {
+	I  Input
+	PI PrivateInput
+}
+
+type Input struct {
+	AgreementStateCommitment frontend.Variable `gnark:",public"`
+	StateObjectCommitment    frontend.Variable `gnark:",public"`
+	CalculatedAgreementRoot  frontend.Variable `gnark:",public"`
+}
+
+type PrivateInput struct {
+	O                  Order
+	OrderRoot          frontend.Variable `gnark:",secret"`
+	OrderSalt          frontend.Variable `gnark:",secret"`
+	BuyerPK            frontend.Variable `gnark:",secret"`
+	AgreementStateRoot frontend.Variable `gnark:",secret"`
+	AgreementStateSalt frontend.Variable `gnark:",secret"`
+}
+
+type Order struct {
+	ProductId frontend.Variable `gnark:",secret"`
+	BuyerSig  frontend.Variable `gnark:",secret"`
+}
+
+var verifyingKey string
+
+func (circuit *OrderCircuit) Define(api frontend.API) error {
+	mimc, err := mimc.NewMiMC(api)
+	if err != nil {
+		fmt.Println(err, "New MiMC instantiation failed")
+		return nil
+	}
+
+	// Asserts Input AgreementStateCommitment and Derived AgreementStateCommitment are equal
+	sum := api.Add(circuit.PI.AgreementStateRoot, circuit.PI.AgreementStateSalt)
+	mimc.Write(sum)
+	api.AssertIsEqual(circuit.I.AgreementStateCommitment, mimc.Sum())
+
+	// Asserts Input StateObjectCommitment and Derived StateObjectCommitment are equal
+	mimc.Reset()
+	sum = api.Add(circuit.PI.OrderRoot, circuit.PI.OrderSalt)
+	mimc.Write(sum)
+	api.AssertIsEqual(circuit.I.StateObjectCommitment, mimc.Sum())
```

Comment:
This also goes back to the naming convention chosen by the mock implementation that preceded this. Changing the naming scheme here would also require changing the naming scheme in the previously committed code. This can be done in a separate pull request for educational purposes, so as to not introduce extra files into this PR.